api_app/api/routes/workspaces.py

```diff
@@ -229,6 +229,16 @@ async def create_workspace_service(response: Response, workspace_service_input:
         logging.error(f"User not authorized to use template: {e}")
         raise HTTPException(status_code=status.HTTP_403_FORBIDDEN, detail=str(e))
 
+    # if template has address_space get an address space
+    if resource_template.properties.get("address_space"):
+        # check workspace has address_spaces property
+        if not workspace.properties.get("address_spaces"):
+            raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=strings.WORKSPACE_DOES_NOT_HAVE_ADDRESS_SPACES_PROPERTY)
+        workspace_service.properties["address_space"] = workspace_repo.get_address_space_based_on_size(workspace_service_input.properties)
+        workspace_patch = ResourcePatch()
+        workspace_patch.properties = {"address_spaces": workspace.properties["address_spaces"] + [workspace_service.properties["address_space"]]}
+        workspace_repo.patch_workspace(workspace, workspace_patch, etag, resource_template_repo, user)
```

Comment:
I think you should explain why you go directly to the repo and bypass the operation (i.e. why not send to RP).

It's confusing since now the API made the decision that the workspace template only "cares" about the first address space in the list which is not generic. Can't it be that a workspace will need all the spaces?
Not sure it's a good thing to do...

Reply:
I've added the workspace changes to this PR.

Reply:
@tamirkamara added a comment, is that sufficient?